TowerDefense/GameEngine.cs

```diff
@@ -24,41 +20,17 @@ internal GameEngine()
 
         internal void Update(GameTime gameTime)
         {
+            delayedActions.RemoveAll(d => d.Update(gameTime));
```

Comment:
Maybe just extract a `Remove()` method to clean up GameEngine.Update so it's not peppered with that logic